set-based-validation/src/main/java/io/axoniq/dev/samples/command/handler/AccountCommandHandler.java

```diff
@@ -0,0 +1,27 @@
+package io.axoniq.dev.samples.command.handler;
+
+import org.axonframework.commandhandling.CommandHandler;
+import org.axonframework.commandhandling.gateway.CommandGateway;
+import org.springframework.stereotype.Component;
+
+import io.axoniq.dev.samples.api.ChangeEmailAddressCommand;
+import io.axoniq.dev.samples.api.RequestEmailChangeCommand;
+import io.axoniq.dev.samples.command.persistence.EmailRepository;
+
+/**
+ * Command handling component that validates if the email address already exists
+ * Links to Validation in a command handling component in the blog
+ *
+ * @author Yvonne Ceelie
+ */
+@Component
+public class AccountCommandHandler {
+    @CommandHandler
+    public void handle(RequestEmailChangeCommand command, CommandGateway commandGateway, EmailRepository emailRepository){
```

Comment:
Isn't is 'nicer' having the gateway and repository injected into the class instead of the method?

Reply:
This could be done both ways, when having more command handler in this component, I would agree